internal/services/apimanagement/api_management_logger_resource_test.go

```diff
@@ -248,6 +273,67 @@ resource "azurerm_api_management_logger" "test" {
 `, data.RandomInteger, data.Locations.Primary, data.RandomInteger, data.RandomInteger, data.RandomInteger, data.RandomInteger)
 }
 
+func (ApiManagementLoggerResource) managedIdentityEventHub(data acceptance.TestData) string {
+	return fmt.Sprintf(`
+provider "azurerm" {
+  features {}
+}
+
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%d"
+  location = "%s"
+}
+
+resource "azurerm_eventhub_namespace" "test" {
+  name                = "acctesteventhubnamespace-%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  sku                 = "Basic"
+}
+
+resource "azurerm_eventhub" "test" {
+  name                = "acctesteventhub-%d"
+  namespace_name      = azurerm_eventhub_namespace.test.name
+  resource_group_name = azurerm_resource_group.test.name
+  partition_count     = 2
+  message_retention   = 1
+}
+
+resource "azurerm_api_management" "test" {
+  name                = "acctestAM-%d"
+  location            = azurerm_resource_group.test.location
+  resource_group_name = azurerm_resource_group.test.name
+  publisher_name      = "pub1"
+  publisher_email     = "pub1@email.com"
+
+  sku_name = "Consumption_0"
+  identity {
+    type = "SystemAssigned"
+  }
+}
+
+resource "azurerm_api_management_logger" "test" {
+  name                = "acctestapimnglogger-%d"
+  api_management_name = azurerm_api_management.test.name
+  resource_group_name = azurerm_resource_group.test.name
+
+  eventhub {
+    name             = azurerm_eventhub.test.name
+    endpoint_address = "${azurerm_eventhub_namespace.test.name}.servicebus.windows.net"
+    client_id        = "SystemAssigned"
```

Comment:
This isn't what I would have expected as input for a `client_id`, this property might be misnamed?

Reply:
According to the documentation, either the actual clientId if User Assigned or "System Assigned" can be used: https://learn.microsoft.com/en-us/azure/api-management/api-management-howto-log-event-hubs?tabs=bicep#logger-with-system-assigned-managed-identity-credentials

Reply:
Thanks for the clarification @implodingduck.

We haven't seen a client ID behave this way before, after some internal discussion we thing `client_id` should actually be replaced with the standard `identity` block that we have throughout the provider. I'll leave comments within the PR on the changes needed to make that happen